### PR TITLE
Al/rework ilog async

### PR DESCRIPTION
**Pull request by agnesLeroy**
closes: https://github.com/zama-ai/tfhe-rs-internal/issues/693

### Check-list:

* [ ] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Relevant issues are marked as resolved/closed, related issues are linked in the description
* [ ] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

**Comment by agnesLeroy** [comment]
This PR is based on the signed scalar div branch, only the last commit concerns ilog2.

**Comment by pdroalves** [comment]
@agnesLeroy can you change target to `al/signed_scalar_div` so reviewing this PR is easier?

**Comment by agnesLeroy** [comment]
@pdroalves this PR is now up to date with main.

File: tfhe/src/integer/gpu/server_key/radix/ilog2.rs [hunk 1]
```diff
@@ -330,6 +331,21 @@ impl CudaServerKey {
         res
```
Comment (guillermo-oyarzun) [inline tfhe/src/integer/gpu/server_key/radix/ilog2.rs:331]: just for consistency, i think unchecked_trailing_zeros should be calling unchecked_trailing_zeros_async

**Review by guillermo-oyarzun — CHANGES_REQUESTED** [review]

**Review by guillermo-oyarzun — APPROVED** [review]

**Pull request merged**